autoblocks/_impl/configs/models.py

```diff
@@ -1,19 +1,38 @@
 from dataclasses import dataclass
 from typing import Any
-from typing import Dict
+from typing import List
 from typing import Optional
 
+try:
+    import pydantic
 
-@dataclass
-class RemoteConfigResponse:
+    assert pydantic.__version__.startswith("2.")
+except (ImportError, AssertionError):
+    raise ImportError(
+        "The Autoblocks config SDK requires pydantic version 2.x. "
+        "You can install it with `pip install pydantic==2.*`."
+    )
+
+
+class FrozenModel(pydantic.BaseModel):
+    model_config = pydantic.ConfigDict(frozen=True)
+
+
+class Property(FrozenModel):
+    name: str
+    value: Any
+
+
+class RemoteConfigResponse(FrozenModel):
     id: str
     version: str
-    value: Dict[str, Any]
+    properties: List[Property]
 
 
 @dataclass
 class RemoteConfig:
     id: str
     # only one of these should be specified
-    version: Optional[str] = None  # can be a version like "1" or "latest"
+    major_version: Optional[int] = None  # should be a sepcific major version
+    minor_version: Optional[int] = None  # omitted for latest minor version
```

Comment:
can we match prompts here? where minor_version type is `str` and is required

Reply:
i guess that will only work if you make major/minor its own object though. but i still think it would be nicer to make ppl be explicit about the minor version. for prompts you have to do:

```python
mgr = Manager(minor_version="0")

# or

mgr = Manager(minor_version="latest")
```